### ISSUE TITLE
ghost fields when using list type in repeatable fields.

### ISSUE THREAD

**Issue by N6REJ** [joomla/joomla-cms#14556]
### Steps to reproduce the issue
create a set of fields that are list type.
view the fields in admin you will see 2, one functions one doesn't.

### Expected result
standard list form field
as in 3.6.x
![image](https://cloud.githubusercontent.com/assets/1850089/23836625/3faed6b6-0749-11e7-8a88-122d047e998f.png)

### Actual result
2 fields for each list field, one functional one none.
![image](https://cloud.githubusercontent.com/assets/1850089/23836607/059da0e2-0749-11e7-8b3b-0ff363558a40.png)
Also field row is NOT repeated.

### System information (as much as possible)
```
				<field name="fontsChooser"
					   type="Repeatable"
					   icon="list"

					   label="TPL_SERENITY_FONTS_CHOOSER_LABEL"
					   description="TPL_SERENITY_FONTS_CHOOSER_DESCRIPTION"
					   select="TPL_SERENITY_FONTS_CHOOSER_SELECT"
					   default="{'Elements':['body','h1','h2','h3','h4','h5','h6','p','a','i','ol','ul','li','span.company-name','span.company-slogan'],'Function':['Body','h1','h2','h3','h4','h5','h6','Paragraph','links','i','Ordered lists','Unordered Lists','List Element','Company Name','Company Moniker'],'Font':['','','','','','','','','','','','','','',''],'Size':['','','','','','','','','','','','','','',''],'Weight':['','','','','','','','','','','','','','',''],'Style':['','','','','','','','','','','','','','',''],'Color':['','','','','','','','','','','','','','',''],'backgroundColor':['','','','','','','','','','','','','','',''],'backgroundhoverColor':['','','','','','','','','','','','','','',''],'linkColor':['','','','','','','','','','','','','','',''],'hoverlinkColor':['','','','','','','','','','','','','','','']}"
				>
					<fieldset hidden="true" name="fontsChooser_modal" repeat="true">
						<field name="Elements"
							   label="TPL_SERENITY_FONTS_ELEMENTS_LABEL"
							   description="TPL_SERENITY_FONTS_ELEMENTS_DESC"
							   class="medium-input"
							   filter="html"
							   type="text"
							   default=''
						/>
                                               <field name="Weight"
							   label="TPL_SERENITY_FONTS_WEIGHT_LABEL"
							   description="TPL_SERENITY_FONTS_WEIGHT_DESC"
							   type="list"
							   default=''
						>
							<option value=''>- Select -</option>
							<option value="normal">Normal</option>
							<option value="bold">Bold</option>
							<option value="bolder">Bolder</option>
							<option value="lighter">Lighter</option>
							<option value="100">100</option>
							<option value="200">200</option>
							<option value="300">300</option>
							<option value="500">500</option>
							<option value="600">600</option>
							<option value="800">800</option>
							<option value="900">900</option>
							<option value="initial">Initial</option>
							<option value="inherit">Inherit</option>
						</field>
						<field name="Style"
							   label="TPL_SERENITY_FONTS_STYLE_LABEL"
							   description="TPL_SERENITY_FONTS_STYLE_DESC"
							   class="small-input"
							   type="list"
							   default=''
						>
							<option value=''>- Select -</option>
							<option value="normal">Normal</option>
							<option value="italic">Italic</option>
							<option value="oblique">Oblique</option>
							<option value="initial">Initial</option>
							<option value="inherit">Inherit</option>
						</field>

</fieldset>
</field>
```
### Additional comments
I realize repeatable is deprecated but it should still function properly at this point, per George

**Comment by N6REJ** [comment]
@wilsonge here's the issue you asked me to file.

**Comment by Bakual** [comment]
Imho it is deprecated since we never got it to work properly. It has been replaced with subforms instead.

**Comment by N6REJ** [comment]
list worked fine.  Honestly, I didn't/don't have any issues with it in 3.6.5
Deprecated doesn't mean it shouldn't function like it did before.

**Comment by wilsonge** [comment]
We know that the modals inside the view never worked. But list views used to and should continue to as this is supposed to be covered by our b/c pledge

**Comment by Bakual** [comment]
Of course, if it worked before it should continue to work.

**Comment by franz-wohlkoenig** [comment]
i don't get this Issue, two rows are shown and editable:
![bildschirmfoto 2017-04-01 um 16 49 35](https://cloud.githubusercontent.com/assets/8235763/24579744/437196a0-16fb-11e7-8509-54c026076160.JPG)
### System information
Joomla! 3.7.0-beta4 (nightly Build)
macOS Sierra, 10.12.4
Firefox 52 (64-bit)
#### MAMP 4.1.1
- PHP 7.0.15
- MySQLi 5.6.35

**Comment by Bakual** [comment]
@franz-wohlkoenig This report isn't about custom fields. The report is a bit misleading.
It's about the "repeatable" formfield, which can be specified in a form.xml (not as custom field).

**Comment by franz-wohlkoenig** [comment]
Thanks for Info @Bakual

**Comment by N6REJ** [comment]
@Bakual in this particular case its in a template if that matters.

**Comment by N6REJ** [comment]
fixed by pr #15042

**Comment by zero-24** [comment]
Closing as there is a PR thanks for the update.
